Example/Tests/SKYReferenceTests.m

```diff
@@ -0,0 +1,51 @@
+//
+//  SKYReferenceTests.m
+//  SKYKit
+//
+//  Copyright 2017 Oursky Ltd.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import <OHHTTPStubs/OHHTTPStubs.h>
+#import <SKYKit/SKYKit.h>
+
+SpecBegin(SKYReference)
+
+    describe(@"SKYReference", ^{
+        it(@"can be copied", ^{
+            SKYReference *ref = [[SKYReference alloc]
+                                 initWithRecordID:[SKYRecordID recordIDWithRecordType:@"book"]];
+            SKYReference *refClone = [ref copy];
+
+            expect(refClone.recordID).to.equal(ref.recordID);
+            expect(refClone.referenceAction).to.equal(ref.referenceAction);
+        });
+
+        it(@"can be encoded and decoded", ^{
+            SKYReference *ref = [[SKYReference alloc]
+                                 initWithRecordID:[SKYRecordID recordIDWithRecordType:@"book"]];
+
+            NSData *data = [NSKeyedArchiver archivedDataWithRootObject:ref];
+            SKYReference *refClone = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+
+            //// Don't know why this cannot pass
+            // expect(refClone.recordID).to.equal(ref.recordID);
```

Comment:
I don’t know too, if I am not mistaken, the runtime consider the two objects have different classes, causing the the equality comparison to fail. This happens after archiving.

Reply:
> the runtime consider the two objects have different classes
That is also what I get during my debugging. However, both `[refClone.recordID class]` and `[ref.recordID class]` return `SKYRecordID` when I print them in lldb console.

Reply:
the object’s `isa` field is different, god knows why